core/src/main/java/org/commonjava/maven/ext/core/ManipulationManager.java

```diff
@@ -118,6 +120,34 @@ public void init( final ManipulationSession session )
     {
         logger.debug( "Initialising ManipulationManager with user properties {}", session.getUserProperties() );
 
+        // We invert it as the property is to _enable_ deprecated properties - which is off by default.
+        final boolean deprecatedDisabled = !Boolean.parseBoolean(session.getUserProperties().getProperty( DEPRECATED_PROPERTIES,
+                "false"));
+
+        session.getUserProperties().stringPropertyNames().forEach(
+                p -> {
+                    if (!p.equals("maven.repo.local") && ConfigList.allConfigValues.keySet().stream().noneMatch(p::startsWith)) {
+                        logger.warn("Unknown configuration value {}", p);
+                    }
+                    // If the configuration value 'x' is deprecated remove it from the property list _if_ deprecated properties
+                    // are NOT enabled. We have to do a rather ugly starts with instead of direct keying as some properties
+                    // operate upon a prefix basis.
+                    if (deprecatedDisabled) {
+                        logger.debug ("Examining for deprecated properties in {}", p);
+                        ConfigList.allConfigValues.entrySet().stream().
+                                filter( e -> p.startsWith(e.getKey())).
+                                filter( Map.Entry::getValue ).
+                                forEach(up ->
+                                {
+                                    logger.warn
+                                            ("Removing key {} (as deprecated) from user properties (with matcher of {})",
+                                                    p,
+                                                    up.getKey());
+                                    session.getUserProperties().remove(p);
+                                });
+                    }
+                });
```

Comment:
A suggestion here is to print (all) warnings and then fail. At this point, user has two options to fix it, enable the deprecated options, or fix that particular option that has the warning. If we had access to a map that had equivalent non-deprecated options, we could suggest that option.

Reply:
It now throws an exception below.